lib.php

```diff
@@ -264,3 +266,100 @@ function add_event_to_sent_log($event) {
         $DB->insert_record('logstore_xapi_sent_log', $newrow);
     }
 }
+
+/**
+ * Extract events from logstore_xapi_log or logstore_xapi_failed_log.
+ *
+ * @return array
+ */
+function logstore_xapi_extract_events($limitnum = 0, $log = XAPI_REPORT_SOURCE_LOG, $type = null) {
+    global $DB;
+
+    $conditions = null;
+    if ($type) {
+        $conditions = array("type" => $type);
+    }
+    $sort = '';
+    $fields = '*';
+    $limitfrom = 0;
+
+    if ($log == XAPI_REPORT_SOURCE_LOG) {
+        $events = $DB->get_records('logstore_xapi_log', $conditions, $sort, $fields, $limitfrom, $limitnum);
+    } else {
+        $events = $DB->get_records('logstore_xapi_failed_log', $conditions, $sort, $fields, $limitfrom, $limitnum);
+    }
+    return $events;
+}
+
+/**
+ * Get event ids.
+ *
```

Comment:
Missing param docs

Reply:
I have added the @param and @return types.